### DISCUSSION TITLE
Allow styling / images for SSO buttons

### DISCUSSION THREAD

**Discussion by jeremybradbury** [directus/directus#12490] [Feature Requests]
The previous thread got locked, but it's still a legal problem for all Directus site maintainers. 

I can give the highlights.

I presented legal issues of not allowing us to use custom button images or styles for SSO. 

In the developer and advertising agreements with every major brand, they require you use specific buttons with specific designs.

A developer dismissed my claims with:

>The buttons on the login page are configured through the env variables, so there's no requirement to use any specific logo or name.

However, this is only partly true. We can remove the image and display text only, however that's still against brand rules outlined by Google and others. We can also not display anything but the proper brand text as it would break the config using another provider name.

I have a way to hide the Directus login page (with the improperly formatted login buttons on it) and use my own instead. This is currently the only workaround... and what Meta advised us to do on our last call.

Some possible solutions include: 

1. move most of the config to the database instead of .env (all but the client id & secret?).
1. allow images to be placed in folder like google.png microsoft.jpg (however if you do the above, we can use the settings UI over naming convention).
1. allow us to create google.css and twitch.css to be included with their auth buttons (again we can make an editble style tag in the db if step 1 happens)

so a folder like /sso-providers/ containing google.png and google.css is the fast fix, no real changes to config api needed.

moving params like those and default role per provider etc can always wait until later.

more details here: https://github.com/directus/directus/discussions/12121

**Comment by benhaynes** [comment]
Option 3 feels like the cleanest and most "Directus" way of doing things, but there's a much easier way to give this the necessary escape hatch. SSO buttons could be given clear and distinct CSS selectors, and then they could be overridden using **Settings > Project Settings > CSS Overrides**.

That said, I want to be transparent that this likely won't bubble up to the docket until it has a meaningful "upvoting" from the community. So a PR/sponsorship is likely the best way to get this implemented in a timely manner.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
